### DISCUSSION TITLE
Circle QskPushButton problem

### DISCUSSION THREAD

**Discussion by hestrro** [uwerat/qskinny#448] [Q&A] [answered]
I want to create a circular button with only ico. When the length and width are less than 48, the ico of the button will be offset from the center. Can this problem be solved?
 auto shapeButton3=new QskPushButton(this);
        shapeButton3->setIconSource("sports_soccer");
        shapeButton3->setPlacementPolicy(QskPlacementPolicy::Ignore);
        shapeButton3->setGeometry(250,320,40,40);
        auto metric3=new QskBoxShapeMetrics(20,20,20,20);
        shapeButton3->setMarginHint(QskPushButton::Icon,10);
        shapeButton3->setShape(*metric2);

**Comment by uwerat** [comment]
Maybe: button->setAlignmentHint( QskPushButton::,Icon, Qt::AlignCenter );

**Comment by hestrro** [comment]
This method is unsuccessful, maybe it still occupies some space when the text is empty. Just like a smaller round button like Google Chrome. When the length and width are less than 48, it will be offset

**Comment by uwerat** [comment]
Please provide a minimal compilable example that demonstrates the problem.

**Comment by hestrro** [comment]
You can create a button like this ：
 
 auto shapeButton3=new QskPushButton(this);
        shapeButton3->setIconSource("sports_soccer");
        shapeButton3->setPlacementPolicy(QskPlacementPolicy::Ignore);
        shapeButton3->setGeometry(250,340,40,40);
        auto metric3=new QskBoxShapeMetrics(20,20,20,20);
        shapeButton3->setAlignmentHint(QskPushButton::Icon,Qt::AlignCenter);
        shapeButton3->setShape(*metric2);

**Comment by uwerat** [comment, marked as answer]
auto button = new QskPushButton();
    button->setShape( { 100, Qt::RelativeSize } );
    button->setPaddingHint( QskPushButton::Panel, 0 );
    button->setFixedSize( 40, 40 );

**Comment by uwerat** [comment]
Because of the default padding from the skin there was no space left for the icon and it ended up somewhere

**Comment by uwerat** [comment]
Note, that setting a relative size of 100% will always create a round button - regardless of the final button size. Setting the radii to 20 pixels will only work for a size of 40x40.

**Comment by uwerat** [comment]
QskControl::setFixedSize is a shortcut for setting the preferredSize and the sizePolicy to Fixed/Fixed. That means that layouts will set the button to the preferred size.

However if you don't have any layout controlling your button you have to set the geometry manually - like in your code snippet.
